scripts/build-images.sh

```diff
@@ -38,23 +40,23 @@ else
 fi
 
 echo "Starting image build"
-export REGISTRY=docker.io/
-export REPO=multicloudlab
+export REGISTRY=quay.io/cluster-api-provider-ibmcloud
 
-echo "Logging into registry ${REGISTRY///}"
-docker login -u "${DOCKER_USERNAME}" -p "${DOCKER_PASSWORD}"
-
-echo "Building controller docker image"
+echo "Building controller and clusterctl docker images."
 cd ${base_dir}
-make container IMG=${REGISTRY}${REPO}/cluster-api-provider-ibmcloud-controller:${TAG}
+VERSION=$TAG make images
 cd -
 
-echo "Pushing image with tag '${TAG}'."
-docker push ${REGISTRY}${REPO}/cluster-api-provider-ibmcloud-controller:${TAG}
+echo "Logging into registry ${REGISTRY%%/*}"
+docker login -u "${REGISTRY_USERNAME}" -p "${REGISTRY_PASSWORD}"
```

Comment:
How to enable all contributors can `push` image to this repo?

Reply:
This PR helps to manually do.

For all contributors, how to make `passwd` security is an issue. Travis CI can make it configured in web site but not exported to any output. The CI script will automatically do it for any users.

However, test-infra needs upload files to the repo. I did not see any way to make `password` private. So, it is still under investigation.

Reply:
just clarify my understanding , this will be used `only` when developer satisfy their development image and utilize this script to upload to quay, right? if this is the case, then keep password local should be fine? 

Forgive me if I understand this incorrect, do we have a use case to let all users push image to the repo? I think we only let infra/travis to CI test and push the image after code merge trigger (at least that's our existing process on other project, not k8s related ) , any developer should not push image to this repo, they can push to their own repo if need test?

Reply:
Yes. We use this script to build and upload images to `quay.io`

We can execute it manually, or we can make anything else to execute it automatically.

For automatically case, e.g. the script can be invokded in `travis` CI for automation:
    - if release `tag` cut, `travis` will automatically upload release images and override `latest` image with latest release
    - if any branch test succeed, `travis` can upload `canary` image
So, automatically execution is convenient. The issue is keep `passwd` security.

Reply:
Ideally, this should be done by CI, if CI passed, push a image with tag `canary`.

Reply:
BTW: @xunpan perhaps you can also check what other projects is doing for this, like aws, vsphere provider etc.

Reply:
They do not have such automatically image upload feature based on my understanding.

Reply:
ok, per above comments, let's merge this and keep enhancing